tests/test_simultaneous_initiate.c

```diff
@@ -1486,6 +1489,245 @@ START_TEST(test_repeated_simultaneous_initiate_lost_message_repeated_messages)
+    ck_assert_int_eq(result, 0);
+
+    session_cipher *bob_session_cipher = 0;
+    result = session_cipher_create(&bob_session_cipher, bob_store, &alice_address, global_context);
+    ck_assert_int_eq(result, 0);
+
+    /* Process the pre key bundles */
+    result = session_builder_process_pre_key_bundle(alice_session_builder, bob_pre_key_bundle);
+    ck_assert_int_eq(result, 0);
+
+    result = session_builder_process_pre_key_bundle(bob_session_builder, alice_pre_key_bundle);
+    ck_assert_int_eq(result, 0);
+
+    /* Alice creates cA */
+    ciphertext_message *alice_cA = 0;
+    result = session_cipher_encrypt(alice_session_cipher, alice_kA, DJB_KEY_LEN, &alice_cA);
+    ck_assert_int_eq(result, 0);
+
+    /* Bob creates cB */
+    ciphertext_message *bob_cB = 0;
+    result = session_cipher_encrypt(bob_session_cipher, bob_kB, DJB_KEY_LEN, &bob_cB);
+    ck_assert_int_eq(result, 0);
+
+    /* Verify message types */ 
+    ck_assert_int_eq(ciphertext_message_get_type(alice_cA), CIPHERTEXT_PREKEY_TYPE);
+    ck_assert_int_eq(ciphertext_message_get_type(bob_cB), CIPHERTEXT_PREKEY_TYPE);
+
+    /* Alice makes cA a pre_key_signal_message */
+    pre_key_signal_message *alice_cA_copy = 0;
+    result = pre_key_signal_message_copy(&alice_cA_copy, (pre_key_signal_message *)alice_cA, global_context);
+    ck_assert_int_eq(result, 0);
+
+    /* Bob makes cB a pre_key_signal_message */
+    pre_key_signal_message *bob_cB_copy = 0;
+    result = pre_key_signal_message_copy(&bob_cB_copy, (pre_key_signal_message *)bob_cB, global_context);
+    ck_assert_int_eq(result, 0);
+
+    /* Alice decrypts cB to get kB */
+    signal_buffer *cB_decrypted = 0;
+    result = session_cipher_decrypt_pre_key_signal_message(alice_session_cipher, bob_cB_copy, 0, &cB_decrypted);
+    ck_assert_int_eq(result, 0);
+
+    /* Bob decrypts cA to get kA */  
+    signal_buffer *cA_decrypted = 0;
+    result = session_cipher_decrypt_pre_key_signal_message(bob_session_cipher, alice_cA_copy, 0, &cA_decrypted);
+    ck_assert_int_eq(result, 0);
+
+    /* Verify that the messages decrypted correctly */
+    uint8_t *cB_decrypted_data = signal_buffer_data(cB_decrypted);
+    ck_assert_int_eq(memcmp(bob_kB, cB_decrypted_data, DJB_KEY_LEN), 0);
+
+    uint8_t *cA_decrypted_data = signal_buffer_data(cA_decrypted);
+    ck_assert_int_eq(memcmp(alice_kA, cA_decrypted_data, DJB_KEY_LEN), 0);
+
+    /* Alice generates shared key */
+    uint8_t *alice_shared_key = 0;
+    result = generate_shared_key(&alice_shared_key, bob_signed_pre_key, alice_signed_pre_key, alice_kA, cB_decrypted_data);
+    ck_assert_int_eq(result, 0);
+
+    /* Bob generates shared key */
+    uint8_t *bob_shared_key = 0;
+    result = generate_shared_key(&bob_shared_key, alice_signed_pre_key, bob_signed_pre_key, cA_decrypted_data, bob_kB);
+    ck_assert_int_eq(result, 0);
+
+    /* Make sure that Alice and Bob have a shared key */ 
+    ck_assert_int_eq(memcmp(alice_shared_key, bob_shared_key, 64), 0);
```

Comment:
It seems that in real life scenario we won't be able to perform this check as each party only has their shared key not the other party's, so I was wondering if we use the created shared keys (alice_shared_key/bob_shared_key) in the encryption itself after this point?